ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java

```diff
@@ -285,6 +285,15 @@ static void setPurgeUptoSnapshotIndex(RaftProperties properties, boolean shouldP
       setBoolean(properties::setBoolean, PURGE_UPTO_SNAPSHOT_INDEX_KEY, shouldPurgeUptoSnapshotIndex);
     }
 
+    String PURGE_PRESERVE_LOG_NUM_KEY = PREFIX + "purge.preserve.log.num";
+    long PURGE_PRESERVE_LOG_NUM_DEFAULT = 0L;
+    static long purgePreserveLogNum(RaftProperties properties) {
+      return getLong(properties::getLong, PURGE_PRESERVE_LOG_NUM_KEY, PURGE_PRESERVE_LOG_NUM_DEFAULT, getDefaultLog());
+    }
+    static void setPurgePreserveLogNum(RaftProperties properties, int purgePreserveLogNum) {
+      setInt(properties::setInt, PURGE_PRESERVE_LOG_NUM_KEY, purgePreserveLogNum);
```

Comment:
We keep set and get unified, setInt -> setLong

Reply:
Done